tests/dummy/app/routes/application.js

```diff
@@ -10,8 +10,8 @@ export default Ember.Route.extend({
   },
 
   actions: {
-    update(newOrder) {
-      this.set('currentModel.items', a(newOrder));
+    update(model, newOrder) {
+      a(model).set('items', a(newOrder));
```

Comment:
Ah… this is because `model` is a POJO rather than an `Ember.Object`. I guess the simplest thing is to use `Ember.set`:

```js
const { A, set } = Ember;

update(model, newOrder) {
  set(model, 'items', a(newOrder));
}
```